addons/test_mail/tests/test_mail_template.py

```diff
@@ -61,6 +62,25 @@ def setUpClass(cls):
         # Force the attachments of the template to be in the natural order.
         cls.test_template.invalidate_recordset(['attachment_ids'])
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_template_schedule_email(self):
+        scheduled_datetime = datetime.datetime.now() + datetime.timedelta(days=3)
+
+        # schedule the mail in 3 days
+        self.test_template.scheduled_date = '{{datetime.datetime.now() + datetime.timedelta(days=3)}}'
+        mail_id = self.test_template.send_mail(self.test_record.id)
+        mail = self.env['mail.mail'].sudo().browse(mail_id)
+        self.assertEqual(
+            mail.scheduled_datetime.replace(second=0, microsecond=0),
+            scheduled_datetime.replace(second=0, microsecond=0),
+        )
```

Comment:
Seems pretty racey, with no freezegun and `send_mail` being non-trivial it's eminently possible that the clock will go through a time boundary between the two, making this a nondetermininstic test.

Should either freeze time or use `assertAlmostEqual` with a small `delta=timeldeta(...)`.

Reply:
It was planned to use freezegun indeed :)